[CS2113-W15-2] PocketPal

jinxuan-owyong opened the pull request:
PocketPal is a money managing application that can help track one's expenses, optimised for use via a Command Line Interface. Users can take advantage of the input flags for entering entries quickly.

nghochi123 committed:
Merge pull request #135 from nghochi123/master

Fix bug where entry datetime not saved and loaded

adenteo committed:
Refactor parser logic

adenteo committed:
Update Parser.java

adenteo committed:
Update parser logic

Users are required to use `-d` flag when adding new expense now to specify the description

adenteo committed:
Update features in user guide

adenteo committed:
Update user guide

adenteo committed:
Update Parser.java

yuanners99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+#### View Command
+##### Class diagram of view command 
+![ViewCommandClassDiagram.png](./static/viewCommandClassDiagram.png)
```
This image is broken in the developer guide.

yuanners99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+<details>
+   <summary>Sequence diagram</summary>
+
+   <img alt="Entry Endpoint [POST] Sequence Diagram" src="docs/../static/backend/EntryEndpointPostSequence.png" />
```
Another broken image :(

yuanners99 commented on docs/DeveloperGuide.md:
````diff
@@ -1,38 +1,1061 @@
+<details>
+<summary>Example</summary>
+
+  ```java
````
Broken format for code snippets

yuanners99 commented on docs/DeveloperGuide.md:
````diff
@@ -1,38 +1,1061 @@
+<details>
+<summary>Expected output:</summary>
+
+```
````
Wrong format of code snippet

yuanners99 commented on docs/DeveloperGuide.md:
````diff
@@ -1,38 +1,1061 @@
+<details>
+<summary>Expected output:</summary>
+
+```
````
All code snippets are not showing correctly on the DG

yuanners99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
+
+## Testing with sample data (from file)
```
No formatting for this whole chunk of text

yuanners99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
```
The hyperlink at the end of the DG "A list of currently supported commands in PocketPal can be found [here]" to the user guide doesnt work

yuanners99 reviewed (commented):
It is organised nicely in a top-down approach, starting with the overall architecture before getting into the individual commands. The UML and sequence diagrams are very clear and concise, chefs kiss. Just need to double check the image links and code snippets formatting🫡🫡🫡

leonghuenweng committed:
Parser Class : remove magic numbers

yuanners99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+The following sequence diagram shows how the add command work:
+
+![AddCommandSequenceDiagram](static/frontend/commands/AddCommandSequenceDiagram.png)
```
For the instance postReq:Request, there is not method calling it. Maybe can add the method name invoking the instance together with the arrow.

calebcjl replied:
![image](https://user-images.githubusercontent.com/82926766/228766327-e355293c-ae6c-45ef-bc7f-9fd2b5b1e67e.png)
Should the activation bar extend out of the alt block?

yuanners99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+The Sequence Diagram below illustrates the interactions within the `Parser` component when a user inputs the following
+command: `/add McDonalds -c Food -p 10.50`
+
+![ParserSequenceDiagram](static/frontend/parser/ParserSequenceDiagram.png)
```
Can try to add activation bar for all diagrams for consistency. There are also missing colons of instances for many diagrams.

YongbinWang replied:
Can consider adding activation bar especially for creation cause the entity "c:AddCommand". Since it is a constructor. You can also consider having better naming convention for your entities as "c" is not very understandable. I like how you add your user input string to your method calls.

yuanners99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+The structure of the Storage class is as follows:
+
+![StorageClassDiagram](static/backend/storage/StorageClassDiagram.png)
```
There is no context for this diagram included, maybe can explain the use of the class diagram here

Ju-Can replied:
Class diagrams are used to show association between classes. Having class diagram showing only the methods in the class is redundant.

yuanners99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+<!-- @@author leonghuenweng -->
+#### Edit Command
+##### Overall class diagram for editing an Entry
+![EditCommandClassDiagram](static/frontend/commands/EditCommandClassDiagram.png)
```
The ref methods save() is not in the DG.

yuanners99 reviewed (commented):
Sequence diagrams are not really consistent.

leonghuenweng committed:
Merge pull request #145 from leonghuenweng/remove-parser-magic-num

Parser Class : remove magic numbers

YongbinWang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+acknowledgement message is printed to the user.
+
+##### Overall sequence diagram for editing an Entry
+
+![img_1.png](static/frontend/commands/EditCommandSequenceDiagram.png)
+
```
Very well done sequence diagram. I like how you included the number index to better illustrate the sequence of process.

YongbinWang reviewed (commented):
Very detailed and informative developers guide. Well done.

T-Wan-Lin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+<details>
+   <summary>Sequence diagram</summary>
+
+   <img alt="Entries Endpoint [GET] Sequence Diagram" src="docs/../static/backend/EntriesEndpointGetSequence.png" />
```
The sequence diagram here is broken and cannot be viewed, so are most of the diagrams in this section.

calebcjl replied:
![image](https://user-images.githubusercontent.com/82926766/228768459-aead351b-1124-4e80-bc8b-f19e22b43ff6.png)
Some images not being displayed as intended

T-Wan-Lin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+</div> -->
+
+<!-- @@author jinxuan-owyong -->
+## Backend
```
Very detailed explanation on how the backend works. Well done!

Ju-Can commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+Here's a class diagram that shows the core structure of the `Parser` class.
+
+![ParserClassDiagram](static/frontend/parser/ParserClassDiagram.png)
```
The class diagram does not help in clarifying the parser mechanism as it shows only the methods in Parser. The explanation below and sequence diagram are sufficient.

honglinshang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+#### Access all entries available
+##### Get recent or all entries
+
+<details>
+   <summary>Sequence diagram</summary>
+
+   <img alt="Entries Endpoint [GET] Sequence Diagram" src="docs/../static/backend/EntriesEndpointGetSequence.png" />
+
+</details>
+
+`GET /entries`
```
Not sure if it's just on my end, but the image here is not loading properly.

honglinshang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+| `200`       | OK                    | Gson-serialised `List<Entry>`, deserialise with `EntryLogParser::deserialise` |
+| `422`       | Unprocessable Content | -                                                                             |
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
```
Good job including these links, they were very useful in navigating your document!

T-Wan-Lin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+### API
+#### Endpoints
+
+![Endpoints](static/backend/endpoint/EndpointClassDiagram.png)
```
Try to minimise the number of intersecting/criss-crossing arrows in the diagram to make it neater and more readable.

T-Wan-Lin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+The following activity diagram summarizes what happens when a user executes a delete   command:
+
+![DeleteCommandActivityDiagram](static/frontend/commands/DeleteCommandActivityDiagram.png)
```
Such summary diagrams are useful in giving the reader a good summary of what occurs when the command is executed. I would recommend using these diagrams more consistently and placing such diagrams right before you elaborate before going into deeper detail about what happens in each step. This makes it more readable.

Ju-Can commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+View Command object.
+
+##### Implementation
+![ViewCommandSequenceDiagram.png](static/frontend/commands/ViewCommandOverallSequenceDiagram.png)
```
There is only one handing over of controls between ViewCommand and Backend, the use of sequence diagram may not be necessary here.

honglinshang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+We use the `EntryLog` data structure to keep track of the entries entered by the user.
+
+![Data Structure Class Diagram](./static/data/DataStructureClassDiagram.png)
```
The arrow from Entry to EntryLog has a label `0...`, seems like it is an error?

honglinshang replied:
The label for 1 above Category is also kind of hidden

honglinshang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+This project uses a simplified HTTP model, where the frontend sends a `Request` to the backend to perform data-related operations. The backend returns a `Response`, which is then processed by the frontend
+
+![Simplified HTTP Model](static/communication/SimplifiedHTTPClassDiagram.png)
```
Same thing for this diagram, there is a `0...` from RequestParams to Request

T-Wan-Lin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+The following sequence diagram shows how the add command work:
+
+![AddCommandSequenceDiagram](static/frontend/commands/AddCommandSequenceDiagram.png)
+![AddCommandSequenceDiagramReference](static/frontend/commands/AddCommandSequenceDiagramReference.png)
```
In this section and in the previous parser class sequence diagrams, there are activation bars missing. Furthermore, do remember to add in crosses in your sequence diagram to show when the method ends, when the object is deleted or no longer referenced.

honglinshang commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+The Sequence Diagram below illustrates the interactions within the `Parser` component upon initialization of PocketPal, as well as whenever data is being saved.
+
+![StorageSequenceDiagram](static/backend/storage/StorageSequenceDiagram.png) 
+
+<div style="text-align: right;">
+   <a href="#table-of-contents"> Back to Table of Contents </a>
+</div>
```
This sequence diagram is missing activation blocks

honglinshang commented on docs/diagrams/frontend/commands/EditCommandSequenceDiagram.puml:
```diff
@@ -0,0 +1,33 @@
+'EXECUTE
+frontend -> EditCommand ++ : execute(ui, backend)
+EditCommand -> Request ** : Request(PATCH, expenseId)
+activate Request
+return new Request
```
Request can be deactivated as it is no longer used afterwards

honglinshang commented on docs/diagrams/frontend/commands/EditCommandSequenceDiagram.puml:
Only some activation bars are present in this diagram, do remember to add the rest in the future!

honglinshang reviewed (commented):
Overall, a lot of work has been put into this DG! Do remember to fix the broken parts, I think most if not all of the dropdowns are not working. Good job :)

Ju-Can commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+
+The backend uses a simplified RESTful API approach. This allows us to decouple code using the proven industry practices. The following diagram illustrates the relationship between various classes involved in `Backend` as described in the [application architecture](#architecture)
+
+![Backend](./static/backend/BackendClassDiagram.png)
```
Labels between classes like "read/write" and "handle request" should include the direction indicated by the black triangles.
<img width="283" alt="Screenshot 2023-03-30 at 13 42 16" src="https://user-images.githubusercontent.com/79389472/228740608-1c1cc46e-4ad5-484a-9120-aa2ea94a3563.png">

T-Wan-Lin commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+</div> -->
+
+<!-- @@author adenteo -->
+## Instructions for manual testing
```
Formatting of all the test case outputs are incorrect and does not show up properly on the DG. Do check your markdowns.
<img width="1056" alt="Screenshot 2023-03-30 at 1 43 17 PM" src="https://user-images.githubusercontent.com/88520987/228740615-ccbcc2ca-1a2c-401c-8483-299b588fc3dd.png">

Ju-Can reviewed (commented):
In general, the DG is detailed and mostly aids in understanding. The diagrams are mostly apt but could be unnecessary at times.

adenteo committed:
Update Parser.java

calebcjl commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+1. When a user enters a command, the `Frontend` uses `Parser` to resolve the user input via `parseUserInput()`.
+2. Within `parseUserInput()`, the corresponding `parseXCommand()` (`X` is a placeholder for the various command
+   names[^1] e.g. `parseAddCommand()`, `parseDeleteCommand()`.)  is invoked to validate that the user input is in the
+   correct format. Any exceptions will be thrown and their corresponding error messages will be shown to the user via
```
Might be confusing for readers as they might think that any exceptions can be thrown. One suggestion is "Any exceptions thrown will show their corresponding error messages..."

calebcjl commented on docs/DeveloperGuide.md:
```diff
@@ -1,38 +1,1061 @@
+The Sequence Diagram below illustrates the interactions within the `Parser` component when a user inputs the following
+command: `/add McDonalds -c Food -p 10.50`
+
+![ParserSequenceDiagram](static/frontend/parser/ParserSequenceDiagram.png)
```
![image](https://user-images.githubusercontent.com/82926766/228763879-d5b6283e-a78d-43fc-8d9a-cfc2c69d0d93.png)

May be clearer to add an activation bar for addCommand class to show its construction

calebcjl reviewed (commented):
Overall very detailed and clear developer guide. Keep it up!

nghochi123 committed:
Fix storage to save amounts in 2 decimal places

adenteo committed:
Update Javadocs for Parser

nghochi123 committed:
Modify read function in Storage to read in as 2dp

adenteo committed:
Complete test coverage for Parser

adenteo committed:
Merge branch 'master' into parser-refactor

adenteo committed:
Update Parser.java

nghochi123 committed:
Merge pull request #148 from nghochi123/master

Fix storage to save amounts in 2 decimal places

nghochi123 committed:
Merge branch 'master' into parser-refactor

nghochi123 committed:
Merge pull request #149 from adenteo/parser-refactor

Parser refactor

adenteo committed:
Update UserGuide.md

adenteo committed:
Update UserGuide.md

adenteo committed:
Update UserGuide.md

Fix formatting of vertical line

adenteo committed:
Merge pull request #150 from adenteo/master

Fix formatting of vertical line in UG

adenteo committed:
Update UserGuide.md

adenteo committed:
Merge branch 'AY2223S2-CS2113-W15-2:master' into master

adenteo committed:
Merge pull request #151 from adenteo/master

Fix vertical line formatting error

adenteo committed:
Create adenteo.md

adenteo committed:
Update help messages

adenteo committed:
Merge branch 'master' into edit-ui-help

adenteo committed:
Fix checkstyle errors

adenteo committed:
Merge branch 'edit-ui-help' of https://github.com/adenteo/tp into edit-ui-help

adenteo committed:
Merge pull request #153 from adenteo/edit-ui-help

Update help messages

adenteo committed:
Update adenteo.md

kaceycsn committed:
Update view function

kaceycsn committed:
Update UI

Update UI so that total expenditure and total income will be printed in standardised format with two decimal places.

nghochi123 committed:
Add new sequence diagrams for Storage

adenteo committed:
Update adenteo.md

kaceycsn committed:
Update CategoryUtil class

jinxuan-owyong committed:
docs: update user guide

- updated table of contents
- added FAQ section

kaceycsn committed:
Update documentation

jinxuan-owyong committed:
docs: fix user guide formatting and links

jinxuan-owyong committed:
Merge pull request #157 from AY2223S2-CS2113-W15-2/update-docs

Update docs

nghochi123 committed:
Merge branch 'master' into docs

adenteo committed:
Update adenteo.md

adenteo committed:
Merge branch 'master' into aden-ppp-draft

nghochi123 committed:
Merge pull request #154 from nghochi123/docs

Add new sequence diagrams for Storage and modify DG

nghochi123 committed:
Modify Backend to treat invalid file as empty

nghochi123 committed:
Update UserGuide to reflect changes in error handling for storage file

adenteo committed:
Update adenteo.md

adenteo committed:
Revise user guide to increase clarity

slightlyharp commented on src/main/java/pocketpal/frontend/commands/AddCommand.java:
```diff
@@ -0,0 +1,111 @@
+     */
+    public AddCommand(String description, double amount, String category) throws InvalidCategoryException {
+        switch (StringUtil.toTitleCase(category)) {
+        case EntryConstants.CLOTHING:
```
Good use of constant

slightlyharp commented on src/main/java/pocketpal/frontend/ui/UI.java:
```diff
@@ -0,0 +1,191 @@
+        String dateTime = entry.getDateTimeString();
+        return "<" + entryID + ">: " + description +
+                " (" + category + ") - $" + UIUtil.formatPrice(price) +
+                " <<" + dateTime + ">>";
```
Consider simplify the expression. Formulate into a string before returning

slightlyharp commented on src/main/java/pocketpal/frontend/commands/ViewCommand.java:
```diff
@@ -0,0 +1,82 @@
+    }
+
+    public ViewCommand(int numberOfEntriesToView,
+                       Category categoryToView, Double priceToViewMin,
```
Indentation for wrapped lines should be 8 spaces

slightlyharp commented on src/main/java/pocketpal/backend/storage/Storage.java:
```diff
@@ -0,0 +1,227 @@
+        File file = new File(this.filePath);
+        file.getParentFile().mkdirs();
+        file.createNewFile();
+        new FileOutputStream(
```
Do not need to start new line for each input parameter. If the line is too long use line wrapping at appropriate places of the line. Indentation for wrapped lines should be 8 spaces

slightlyharp commented on src/main/java/pocketpal/backend/storage/Storage.java:
```diff
@@ -0,0 +1,227 @@
+            throw new InvalidReadFileException(
+                String.format(
+                    "%s%s",
+                    MiscellaneousConstants.GENERAL_STORAGE_ERROR_MESSAGE,
```
Good use of constant

slightlyharp commented on src/main/java/pocketpal/backend/storage/Storage.java:
```diff
@@ -0,0 +1,227 @@
+            LocalDateTime dateTime = DateTimeUtil.convertStringToLocalDateTime(
+                dateTimeString
+            );
+            return new Entry(
```
Consider simplify the expression by creating the variable first

slightlyharp commented on src/main/java/pocketpal/frontend/parser/Parser.java:
```diff
@@ -0,0 +1,549 @@
+    private static final Pattern PRICE_PATTERN = Pattern.compile("(-p|-price)\\s+(\\S+)");
+    private static final Pattern ID_PATTERN = Pattern.compile("(\\s+)?(\\S+)");
+    private static final Pattern START_DATE_PATTERN = Pattern.compile("(-sd|-startdate)\\s+(0*\\d+/0*\\d+/\\d{2,})");
+    private static final Pattern END_DATE_PATTERN = Pattern.compile("(-ed|-enddate)\\s+(0*\\d+/0*\\d+/\\d{2,})");
```
Consider create a new file for the constants to standardise with other classes.

slightlyharp commented on src/main/java/pocketpal/frontend/ui/UI.java:
```diff
@@ -0,0 +1,191 @@
+     */
+    public void printHelp() {
+        print(MessageConstants.MESSAGE_HELP + MessageConstants.MESSAGE_ADD_COMMAND
+                + MessageConstants.MESSAGE_DELETE_COMMAND + MessageConstants.MESSAGE_EDIT_COMMAND
```
Good use of constant for the print

slightlyharp reviewed (commented):
Well done. Good effort in writing the javadoc comments.

jinxuan-owyong committed:
Merge branch 'master' into fix-read-file-exception

jinxuan-owyong committed:
fix: output all logs to file

corrupt file data results in logs being output when `Backend` is
initialised.
this happens prior to logging output to file being setup,
resulting in logs being shown in the CLI.
instead, all logging is now output after logging setup is complete.

jinxuan-owyong committed:
ci: only run coverage check on Linux

jinxuan-owyong committed:
Merge pull request #204 from nghochi123/master

Modify Backend to treat invalid file as empty

jinxuan-owyong committed:
Merge pull request #156 from kaceycsn/fix-read-file-exception

Fix read file exception

jinxuan-owyong committed:
Merge branch 'master' into fix-total-expenditure

jinxuan-owyong committed:
Merge pull request #207 from AY2223S2-CS2113-W15-2/ci-update-workflow

ci: only run coverage check on Linux

nghochi123 committed:
Add unit test for assertion

adenteo committed:
Update UserGuide.md

nghochi123 committed:
Fix unformatted code in DeveloperGuide

kaceycsn committed:
Add unit tests

nghochi123 committed:
storage: add logger and extract string to double parser code to function

nghochi123 committed:
Modify static variables in Storage

jinxuan-owyong committed:
endpoint: add javadoc and logging for methods

jinxuan-owyong committed:
Backend: improve code defensiveness

- only up to 9 digits are allowed for amount, with optional 2 decimal points
- introduce utility class for validating parameters from requests
- use boundary value analysis for unit tests
- allow flexible floating point input for amount
- use utility methods for consistent conversion from double to string

adenteo committed:
Initialise new parser classes

adenteo committed:
Refactor parser

adenteo committed:
Restructure parser

Abstracted parser into command parser classes

adenteo committed:
Fix style violations

leonghuenweng committed:
Add Introduction in User Guide

A brief description of the purpose of PocketPal helps users understand
the app better.

jinxuan-owyong committed:
unit tests: improve backend coverage

- improve line coverage for exceptions thrown by endpoints
- renamed MESSAGE_INVALID_PRICE* to MESSAGE_INVALID_AMOUNT* for more accurate representation

jinxuan-owyong committed:
Merge pull request #211 from AY2223S2-CS2113-W15-2/123-bug-backend-does-not-reject-invalid-price-string

`Backend`: improve code defensiveness

jinxuan-owyong committed:
fix: view invalid id

- updated parser default constants if amount start and end are not passed
- update exception message for invalid index/non-positive integers

jinxuan-owyong committed:
Merge pull request #215 from AY2223S2-CS2113-W15-2/fix-view-invalid-id

fix: view invalid id

jinxuan-owyong committed:
Merge branch 'master' into fix-total-expenditure

jinxuan-owyong committed:
Merge branch 'master' into fix-total-expenditure

jinxuan-owyong committed:
Merge pull request #155 from kaceycsn/fix-total-expenditure

Fix total expenditure

jinxuan-owyong committed:
Merge branch 'master' into fix-storage-test

jinxuan-owyong committed:
Merge pull request #208 from nghochi123/fix-storage-test

Add unit test for assertion

leonghuenweng committed:
Add example command usage to help function

Users don't have to constantly refer to the user guide when they are
unfamiliar with the commands.

jinxuan-owyong committed:
Merge branch 'master' into parser-refactor-2

adenteo committed:
Add exception handling for incorrect date in parser

adenteo committed:
Merge branch 'parser-refactor-2' of https://github.com/adenteo/tp into parser-refactor-2

adenteo committed:
Fix code violation error

adenteo committed:
Add exception handling for invalid price

adenteo committed:
Merge branch 'master' into aden-ppp-draft

adenteo committed:
Merge pull request #203 from adenteo/aden-ppp-draft

Draft personal PPP

adenteo committed:
Merge branch 'master' into docs

adenteo committed:
Update UserGuide.md

leonghuenweng committed:
Improve code quality

jinxuan-owyong committed:
Backend: add integration tests

- updated Backend constructor to use correct storage path for tests

jinxuan-owyong committed:
Backend: add warning for invalid save data

adenteo committed:
Add unknown arguments handling

Throw error if there exists any arguments specified in `/add` with no preceding options

adenteo committed:
Merge branch 'master' into parser-refactor-2

kaceycsn committed:
Update coding standard

Use convertStringToCategory() method to shorten code

adenteo committed:
Remove unwanted files

adenteo committed:
Add complete javadoc and logging

nghochi123 committed:
Merge pull request #209 from nghochi123/docs

Fix some DG formatting issues

jinxuan-owyong committed:
Merge branch 'master' into parser-refactor-2

jinxuan-owyong committed:
Merge pull request #217 from adenteo/parser-refactor-2

Complete refactor of parser class

jinxuan-owyong committed:
Backend: return more meaningful error messages

When attempting to access an entry ID that is invalid, user will be notified the expected range of values. Behaviour has been updated for /edit and /delete

leonghuenweng committed:
AboutUs.md : update personal details

Include github username and porfolio link

leonghuenweng.md : update PPP

leonghuenweng committed:
Merge pull request #219 from leonghuenweng/shawn-ppp

AboutUs.md : update personal details

adenteo committed:
Update UserGuide.md

adenteo committed:
Merge branch 'master' into user-guide-revision

adenteo committed:
Merge pull request #206 from adenteo/user-guide-revision

Revise user guide to increase clarity

nghochi123 committed:
Add draft of PPP: nghochi.md

nghochi123 committed:
Merge branch 'master' into master

nghochi123 committed:
Merge pull request #220 from nghochi123/master

Add draft of PPP: nghochi.md

jinxuan-owyong committed:
EntryEndpoint: validate Entry data on POST

leonghuenweng committed:
Merge branch 'master' into add-introduction-in-UG

adenteo committed:
Edit invalid id message

Add additional invalid id message to indicate the invalid id to user clearly

leonghuenweng committed:
Merge pull request #213 from leonghuenweng/add-introduction-in-UG

Add Introduction in User Guide

adenteo committed:
Update MessageConstants.java

adenteo committed:
Update DeveloperGuide.md

adenteo committed:
Merge branch 'master' into add-examples-to-help-function

adenteo committed:
Merge pull request #218 from leonghuenweng/add-examples-to-help-function

Add example command usage to help function

adenteo committed:
Merge branch 'master' into developer-guide-revision

adenteo committed:
Merge pull request #221 from adenteo/developer-guide-revision

Revise Developer guide

leonghuenweng committed:
Add view and help function contributions

Add user and developer guide contributions

kaceycsn committed:
Update sequence diagram

kaceycsn committed:
Merge branch 'master' into update-sequence-diagram-and-add-function

kaceycsn committed:
Resolve conflicts

nghochi123 committed:
Rename PPP

nghochi123 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W15-2/tp

nghochi123 committed:
Merge pull request #224 from nghochi123/master

Rename PPP

jinxuan-owyong committed:
Merge branch 'master' into testing-improve-coverage

jinxuan-owyong committed:
fix: code quality check

adenteo committed:
Update parser diagrams

jinxuan-owyong committed:
Merge pull request #223 from AY2223S2-CS2113-W15-2/testing-improve-coverage

improve code defensiveness and coverage

adenteo committed:
Update AboutUs.md

adenteo committed:
Merge branch 'master' into parser-dg-revision

adenteo committed:
Merge pull request #225 from adenteo/parser-dg-revision

Parser dg revision

adenteo committed:
Achieve complete test coverage for parser

adenteo committed:
Merge branch 'master' into parser-code-coverage

adenteo committed:
Edit assert statements

adenteo committed:
Merge branch 'parser-code-coverage' of https://github.com/adenteo/tp into parser-code-coverage

kaceycsn committed:
Update help message

adenteo committed:
Merge pull request #226 from adenteo/parser-code-coverage

Achieve complete test coverage for parser

adenteo committed:
Update adenteo.md

adenteo committed:
Merge pull request #228 from adenteo/master

Update adenteo.md

jinxuan-owyong committed:
EntryLog: sort entries by date

- All entries will be shown in order of ascending date when EntryLog is constructed
- Guards against manipulation of saved date/time
- Allows for future expansion for sorting in reverse date order

leonghuenweng committed:
Merge branch 'master' into shawn-ppp

leonghuenweng committed:
Merge pull request #229 from leonghuenweng/shawn-ppp

Add view and help function contributions to PPP

jinxuan-owyong committed:
Entry: improve code readability

- Add documentation and logging for non-trivial methods
- Simplified getCategoryString(), convertStringToCategory() and testGetCategoryString() to reduce code repetition

leonghuenweng committed:
Remove redundant png image under /docs

leonghuenweng committed:
Merge pull request #230 from leonghuenweng/remove-redundant-png

Remove redundant png image under /docs

jinxuan-owyong committed:
Merge branch 'master' into entrylog-improvements

kaceycsn committed:
Merge branch 'master' into update-sequence-diagram-and-add-function

kaceycsn committed:
Merge pull request #222 from kaceycsn/update-sequence-diagram-and-add-function

Update sequence diagram and add function

jinxuan-owyong committed:
Merge pull request #231 from AY2223S2-CS2113-W15-2/entrylog-improvements

Entrylog improvements

leonghuenweng committed:
Add link to SimpleDateTime implementation to address issue #194

Users are advised to set start and end date with a maximum of 5 years
duration in between.

leonghuenweng committed:
Merge pull request #233 from leonghuenweng/add-explanation-issue194

Add link to SimpleDateTime implementation to address issue #194

kaceycsn committed:
Add categories to help message

kaceycsn committed:
Update message constants

kaceycsn committed:
Update message constants

adenteo committed:
Update ParserConstants.java

Fix bug where `-description` option reads `escription` as input

adenteo committed:
Merge branch 'master' into master

adenteo committed:
Update ParserConstants.java

adenteo committed:
Merge branch 'master' of https://github.com/adenteo/tp

kaceycsn committed:
Update help command

Update help command to display more help for each specific command.

Add to help command such that a more detailed guide will be displayed when user enters /help add , /help edit etc.

kaceycsn committed:
Edit unit tests

kaceycsn committed:
Merge branch 'master' into add-categories-to-help-function

kaceycsn committed:
Add Junit tests

Add unit tests for help command and UI

kaceycsn committed:
Improve code coverage

jinxuan-owyong committed:
Merge pull request #240 from adenteo/master

Update ParserConstants.java

kaceycsn committed:
Update help message and remove unused enum

Remove unused Command enum class.

Update help message constants to be more detailed

kaceycsn committed:
Update unit test

kaceycsn committed:
Fix bug

Fix bug when double spaces is input in the help command

leonghuenweng committed:
Modify VALID_PRICE_REGEX in ParserConstants Class

VALID_PRICE_REGEX now only matches with [1-9 digits] . [<= 2 dp].

Add more tests in ParserTest Class for View, Edit and Add commands

To ensure that all invalid amount arguments are rejected regardless of
the command type.

Modified MESSAGE_INVALID_AMOUNT in MessageConstants Class

To give the users a more accurate description of the cause of this
error.

leonghuenweng committed:
Increase test coverage of checkPriceValidity method in ParseCommand

Increase test coverage of execute method in DeleteCommand Class

adenteo committed:
Update parser class diagram

Update `Parser` class diagram to include `Frontend`, `Backend` and `UI`

jinxuan-owyong committed:
docs: add sequence diagram for main components

jinxuan-owyong committed:
docs: add backend changes to devguide

jinxuan-owyong committed:
Merge pull request #243 from AY2223S2-CS2113-W15-2/jx-update-docs-v2.1

update developer guide

leonghuenweng committed:
Merge branch 'master' into fix-auto-rounding-beyond-2dp

leonghuenweng committed:
Merge pull request #242 from leonghuenweng/fix-auto-rounding-beyond-2dp

Modify VALID_PRICE_REGEX in ParserConstants Class

adenteo committed:
Merge branch 'AY2223S2-CS2113-W15-2:master' into master

leonghuenweng committed:
Modify checkDateValidity method in ParseCommand Class

date strings that don't strictly match dd/mm/yy format will trigger the
InvalidDateException.

leonghuenweng committed:
Merge pull request #244 from leonghuenweng/handle-invalid-date

Modify checkDateValidity method in ParseCommand Class

kaceycsn committed:
Update user guide

kaceycsn committed:
Update unit tests

adenteo committed:
Revise expense naming convention

Revise naming of expenses/expenditure to accommodate addition of income

kaceycsn committed:
Update developer guide

kaceycsn committed:
Merge branch 'master' into add-categories-to-help-function

adenteo committed:
Align all files with new year format

Edit all files to align with new `yyyy` format

adenteo committed:
Merge branch 'master' of https://github.com/adenteo/tp

adenteo committed:
Merge branch 'master' into master

kaceycsn committed:
Update ppp

kaceycsn committed:
Improve code quality

leonghuenweng committed:
Add additional constraints on the prices that users supply

Mention that negative numbers will trigger unknown option exception.
Emphasize that all prices supplied must have at most 2 DP accuracy.

leonghuenweng committed:
Fix minor errors such as invalid image path and minor grammatical errors

leonghuenweng committed:
Merge branch 'master' into update-user-guide

leonghuenweng committed:
Merge branch 'master' into update-developer-guide

leonghuenweng committed:
Merge pull request #246 from leonghuenweng/update-user-guide

Add additional constraints on the prices that users supply

leonghuenweng committed:
Merge pull request #247 from leonghuenweng/update-developer-guide

Update developer guide

adenteo committed:
Merge branch 'master' into expense-renaming

adenteo committed:
Merge pull request #245 from adenteo/expense-renaming

Align code with new date format

adenteo committed:
Achieve 100% test coverage for Parser

adenteo committed:
Merge pull request #248 from adenteo/parser-final-revisions

Achieve 100% test coverage for Parser

adenteo committed:
Add NFR

nghochi123 committed:
Modify DG to include value proposition and user stories

nghochi123 committed:
Merge branch 'master' of https://github.com/AY2223S2-CS2113-W15-2/tp

nghochi123 committed:
Modify TOC

nghochi123 committed:
Merge pull request #249 from nghochi123/master

Modify DG to include value proposition and user stories

jinxuan-owyong committed:
docs: update user guide formatting

kaceycsn committed:
Merge branch 'master' into add-categories-to-help-function

jinxuan-owyong committed:
docs: update portfolio path in aboutus.md

kaceycsn committed:
Merge pull request #227 from kaceycsn/add-categories-to-help-function

Improve clarity of help message

jinxuan-owyong committed:
Revert "Improve clarity of help message"

jinxuan-owyong committed:
Merge pull request #251 from AY2223S2-CS2113-W15-2/revert-227-add-categories-to-help-function

Revert "Improve clarity of help message"

jinxuan-owyong committed:
Merge branch 'master' into jx-update-docs-v2.1

jinxuan-owyong committed:
docs: update reposense tags in dg

kaceycsn committed:
Update ppp

nghochi123 committed:
Format Product Scope section

nghochi123 committed:
Merge pull request #252 from nghochi123/master

Format Product Scope section

kaceycsn committed:
Revert "Revert "Improve clarity of help message""

This reverts commit 551340ca490abc079159b3206bf8197fb80740e5.

kaceycsn committed:
Merge branch 'updated-tm' into add-categories-to-help-copy

kaceycsn committed:
Revert overwritten message constants

kaceycsn committed:
Revert overwritten message constants

kaceycsn committed:
Update ppp

kaceycsn committed:
Revert overwritten changes

adenteo committed:
Merge pull request #254 from kaceycsn/add-categories-to-help-copy

Improve help function clarity

adenteo committed:
Merge branch 'master' into dg-enhancement-aden

leonghuenweng committed:
Update ViewCommand sequence diagram

leonghuenweng committed:
Merge branch 'master' into update-viewCommand-sequenceDiagram

jinxuan-owyong committed:
Merge branch 'master' of github.com:AY2223S2-CS2113-W15-2/tp into jx-update-docs-v2.1

leonghuenweng committed:
Merge pull request #256 from leonghuenweng/update-viewCommand-sequenceDiagram

Update ViewCommand sequence diagram

kaceycsn committed:
Update ppp

kaceycsn committed:
Merge branch 'master' into update-ppp-kacey

kaceycsn committed:
Merge pull request #257 from kaceycsn/update-ppp-kacey

Update ppp kacey

adenteo committed:
Merge branch 'master' into dg-enhancement-aden

adenteo committed:
Merge pull request #255 from adenteo/dg-enhancement-aden

Add NFR

jinxuan-owyong committed:
EntriesEndpoint: remove unthrown exception

jinxuan-owyong committed:
docs: add PPP (jinxuan-owyong.md)

jinxuan-owyong committed:
Merge branch 'master' into jx-update-docs-v2.1

jinxuan-owyong committed:
Merge pull request #258 from AY2223S2-CS2113-W15-2/jx-update-docs-v2.1

update docs

jinxuan-owyong committed:
docs: add page break

jinxuan-owyong committed:
Merge branch 'jx-update-docs-v2.1' of github.com:AY2223S2-CS2113-W15-2/tp into jx-update-docs-v2.1

jinxuan-owyong committed:
Merge pull request #259 from AY2223S2-CS2113-W15-2/jx-update-docs-v2.1

docs: add page break

jinxuan-owyong committed:
docs: update devguide headings

jinxuan-owyong committed:
docs: update testing

adenteo committed:
Update adenteo.md

adenteo committed:
Merge branch 'master' into master

adenteo committed:
Merge pull request #260 from adenteo/master

Update adenteo.md

jinxuan-owyong committed:
Merge branch 'master' into jx-update-docs-v2.1

jinxuan-owyong committed:
Merge pull request #261 from AY2223S2-CS2113-W15-2/jx-update-docs-v2.1

docs: update devguide

jinxuan-owyong committed:
Update UserGuide.md

jinxuan-owyong committed:
Update UserGuide.md

fix broken links

jinxuan-owyong committed:
Update DeveloperGuide.md
